libraries/AP_HAL/Util.cpp

```diff
@@ -77,6 +77,23 @@ uint64_t AP_HAL::Util::get_system_clock_ms() const
 #endif
 }
 
+uint64_t AP_HAL::Util::get_system_clock_us() const
+{
+#if defined(__APPLE__) && defined(__MACH__)
```

Comment:
this looks quite odd doing ifdefs for system time inside AP_HAL/Util.cpp. Normally we'd make it a virtual function and implement in each HAL.
I actually think this isn't really the job of the HAL though, I think we need a AP_RTClock library for this work

Reply:
This is the exactly same code that exist in `uint64_t AP_HAL::Util::get_system_clock_ms()`. I disagree that is necessary an entire library to hold this simple PR.